src/main/java/org/spongepowered/api/service/permission/Subject.java

```diff
@@ -99,6 +124,26 @@
      */
     SubjectReference asSubjectReference();
 
+    /**
+     * Get the game object that may be associated with this subject.
+     *
+     * <p>This could be a player, system subject, or something else. The return
+     * value of this method should not be stored.</p>
+     *
+     * @return a potential game object
+     */
+    Optional<?> associatedObject();
+
+    @Override
+    default Cause contextCause() {
+        final /* @Nullable */ Object associated = this.associatedObject().orElse(null);
```

Comment:
@gabizou would appreciate your input since this is an issue me and luck haven't been able to reach a conclusion on.

There are two main cases for permissions checks, with different contexts applicable for each:

1. Current game state (for protection checks, etc): this works fine using the normal cause tracker
2. State of the subject itself (for notifications, announcements, etc):
   This is a bit trickier, since we have to build a custom cause representing the subject's associated object. I've tried to do that here -- is this at all correct?

Right now I've tried to do different defaults depending on how the cause is queried -- a `CommandCause` uses its stored Cause which represents game state, while a direct query to `Subject` will use the associated object (aiming to be similar to API 7 behaviour). This split is inconvenient, but I think it's the only way to provide behaviour that is close to what developers are looking for most of the time. Do you have any thoughts on a better way to capture/pass around a representation of game state?